sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2218,7 +2217,7 @@ public QueryCompatibilityMode getQueryCompatibilityMode() {
             this,
             ResourceType.Document,
             Document.class,
-            collection.getSelfLink(),
+            collection.getAltLink(),
```

Comment:
@FabianMeiswinkel  is okie to change to use AltLink here?

Reply:
so currently, this is only used to build the request path. Which currently can be name based or not. It will not be passed to any cache etc, so should be safe to change here.